sanity-io/structured-content-2022
# Catch-all page rendering Routes ↔ Landing Pages

# Catch-all page rendering Routes ↔ Landing Pages

## Intent

- Add a `[[...slug].tsx` catch-all page which renders Sanity _Route_ entities and their respective _Landing Page_ entities
- Extracts `CookieConsent` component and rewrites styling in plain CSS
- Deletes TailwindCSS and all "legacy" Tailwind-based pages, removes Tailwind-related dependencies and configuration

## Description

This PR is the result of a [request](https://wja.slack.com/archives/C01CT37P5HC/p1644823800644089) by @kmelve on Slack, calling for replicating the Route ↔ Landing Page separation pattern referenced to in [this Loom](https://www.loom.com/share/49d2263fcd4341e6aca313efbab576e7) and as seen in e.g. the www-sanity-io repo.

Other files defined in _/pages_ will have precedence over the new catch-all route, so e.g. [/about](http://localhost:3000/about) will still render _/web/pages/about.tsx_. This is intentional, both due to the fact that Sanity is currently lacking the corresponding [data for Routes](http://localhost:3333/desk/route), and because <em>Landing Page</em>s seem to only support regular "sections" (as typed out by _/web/types/Section.ts_) rather than the more complex and varying nature of the content in the current routes (<em>/about, /program, /tickets, etc</em>) as required by the wireframes.

Note that this breaks `staging`'s backward compatibility with `main`; after this PR is merged to `staging`, `staging` will no longer be mergable to `main` without breaking the [current home page](https://structured-content-2022.sanity.build/). Having `staging` be our development branch (and only merging it to `main` at the time of the new page's release in its entirety) was the plan all along, but this PR breaks a backward compatibility which `staging` has had until now.
 
## Testing this PR

1. `docker-compose up` in the repository's root
2. Go to [/test](http://localhost:3000/test) and verify that the _Page_ associated with the [test route](http://localhost:3333/desk/route;188f7c30-b8c4-4c46-9917-33722517f9ce) I created is rendered.
3. Go to [/home](http://localhost:3000/home) (and the other pages linked in the nav) and verify that they all still look as awful as they did prior to this PR
4. Open any page in an Incognito window and verify that you get the `<CookieConsent />` banner rendered. Verify that the banner looks and behaves exactly like that in the current ["production"](https://structured-content-2022.sanity.build/) version
